avocado/utils/network/hosts.py

```
@@ -44,73 +48,113 @@ class Host:
     def __init__(self, host):
+        """Initializes the base Host class.
+
+        This constructor is intended to be called by sub-classes.
+        Direct initialization of `Host` will raise a TypeError.
+
+        :param host: The hostname or IP address of the host.
+                    For `LocalHost`, this typically defaults to 'localhost'.
+        :type host: str
+        :raises TypeError: If the `Host` class is instantiated directly.
+        """
         if type(self) == Host:  # pylint: disable=C0123
```

Ruff (0.12.2)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)